react/features/base/media/components/web/Video.js

```diff
@@ -0,0 +1,218 @@
+     * with an object.
+     *
+     * @inheritdoc
+     * @returns {boolean} - False is always returned to blackbox this component.
+     * from React.
+     */
+    shouldComponentUpdate(nextProps) {
+        const currentJitsiTrack = this.props.videoTrack
+            && this.props.videoTrack.jitsiTrack;
+        const nextJitsiTrack = nextProps.videoTrack
+            && nextProps.videoTrack.jitsiTrack;
+
+        if (currentJitsiTrack !== nextJitsiTrack) {
+            this._detachTrack(this.props.videoTrack);
+            this._attachTrack(nextProps.videoTrack);
+        }
+
+        return false;
+    }
+
+    /**
+     * Renders the video element.
+     *
+     * @override
+     * @returns {ReactElement}
+     */
+    render() {
+        // The wrapping div is necessary because temasys will replace the video
+        // with an object but react will keep expecting the video element. The
+        // div gives a constant element for react to keep track of.
+        return (
+            <div>
+                <video
+                    autoPlay = { true }
+                    className = { this.props.className }
+                    id = { this.props.id }
+                    ref = { this._setVideoElement } />
+            </div>
+        );
+    }
+
+    /**
+     * Calls into the passed in track to associate the track with the
+     * component's video element and render video.
+     *
+     * @param {Object} videoTrack - The redux representation of the
+     * {@code JitsiLocalTrack}.
+     * @private
+     * @returns {void}
+     */
+    _attachTrack(videoTrack) {
+        if (!videoTrack || !videoTrack.jitsiTrack) {
+            return;
+        }
+
+        const updatedVideoElement
+            = videoTrack.jitsiTrack.attach(this._videoElement);
+
+        // Sets the instance variable for the video element again as the element
+        // maybe have been replaced with a new object by temasys.
+        this._setVideoElement(updatedVideoElement);
+    }
+
+    /**
+     * Removes the association to the component's video element from the passed
+     * in redux representation of jitsi video track to stop the track from
+     * rendering. With temasys, the video element must still be visible for
+     * detaching to complete.
+     *
+     * @param {Object} videoTrack -  The redux representation of the
+     * {@code JitsiLocalTrack}.
+     * @private
+     * @returns {void}
+     */
+    _detachTrack(videoTrack) {
+        // Detach the video element from the track only if it has already
+        // been attached. This accounts for a special case with temasys
+        // where if detach is being called before attach, the video
+        // element is converted to Object without updating this
+        // component's reference to the video element.
+        if (this._videoElement
+            && videoTrack
+            && videoTrack.jitsiTrack
+            && videoTrack.jitsiTrack.containers.includes(this._videoElement)) {
```

Comment:
I know that this line is the old code just moved to this file but I just wanted to ask:

Why do we have this check here? If the track is not attached previously, `detach` won't do anything. IMHO even if we want this check probably we should put it in lib-jitsi-meet. Also I don't think jitsiTrack.containers should be part of the public API of lib-jitsi-meet since it creates/updates it internally. Although I agree that is very hard to tell what is the public API of lib-jitsi-meet.

I don't think this is a problem of this PR. If we decide to change the line somehow we should do it trough another PR probably.

Reply:
Not 100% sure but the comment above mentions an edge case about temasys.

Reply:
Yes but probably this should be included in lib-jitsi-meet if it is not right now.